### PR TITLE
Put group labels on teacher view [#154488491]

### PR DESCRIPTION
**Pull request by kswenson**

### CODE REVIEW

File: src/code/views/teacher-view.tsx [hunk 1]
```diff
@@ -161,10 +175,18 @@ export class TeacherView extends React.Component<
                       simulation.stations.getStation(cell.weatherStationId);
       const precip = precipDiv(station);
       const city = showCities ? cityAnnotation(cell.weatherStationId) : null;
+      const cellLabel = cellName(cell.row, cell.column);
+      const group = groupMap[cellLabel];
+      const groupLabel = group
+                          ? <div className="grid-cell-group-label" style={styles.groupLabel}>
+                              {group || ''}
```
Comment (knowuh) [inline src/code/views/teacher-view.tsx]: At first glance this `||` check is redundant.

File: src/code/views/teacher-view.tsx [hunk 2]
```diff
@@ -147,8 +153,16 @@ export class TeacherView extends React.Component<
...
+          groupMap: { [index:string]: string } = {};
 
+    if (simulation) {
+      simulation.presences.presenceList.forEach((presence) => {
+        if (presence.groupName && presence.weatherStationID) {
+          groupMap[presence.weatherStationID] = presence.groupName;
+        }
+      });
+    }
     const colorFunc = (cell:IGridCell) => {
```
Comment (knowuh) [inline src/code/views/teacher-view.tsx]: Once upon a time `colorFunc` only computed the color of cells. Its been on my mind to rename it someday.  ;)

File: src/code/views/teacher-view.tsx (hunk 1 above)
Comment (kswenson) [reply src/code/views/teacher-view.tsx]: Good catch; removed.

File: src/code/views/teacher-view.tsx (hunk 2 above)
Comment (kswenson) [reply src/code/views/teacher-view.tsx]: Seems beyond the scope of this PR, so I"ll save it for another time.

**Pull request merged**